spec/support/spree/config_helper.rb

```diff
@@ -0,0 +1,18 @@
+module Spree
+  module ConfigHelper
+    def with_spree_config(hash)
+      stored = config_slice(hash.keys)
+      Config.set(hash)
+      yield
+      Config.set(stored)
+    end
+
+    private
+
+    def config_slice(keys)
+      keys.each_with_object({}) do |key, hash|
+        hash[key] = Config[key]
```

Comment:
So if a key doesn't exist, it gets overwritten with `nil`? Isn't that different to a non-existent key?

I'm wondering if we can replace the whole config hash with a copy instead of overwriting values. But I'm not familiar with the Spree config code.

Reply:
Good pickup. That's a bug.

Could you explain your 'copy' idea a bit better?

Reply:
You mean like mocking the whole Config? I suppose we could try that....

Reply:
Yes, but I don't really know how Config works internally.